interop-cats/jvm/src/main/scala/scalaz/zio/interop/catsjvm.scala

```diff
@@ -273,6 +300,57 @@ private class CatsAlternative[R, E: Monoid] extends CatsMonadError[R, E] with Al
   override final def empty[A]: ZIO[R, E, A] = raiseError(Monoid[E].empty)
 }
 
+private class CatsZManagedMonad[R, E] extends Monad[ZManaged[R, E, ?]] {
+  override def pure[A](x: A): ZManaged[R, E, A] = ZManaged.succeed(x)
+
+  override def flatMap[A, B](fa: ZManaged[R, E, A])(f: A => ZManaged[R, E, B]): ZManaged[R, E, B] = fa.flatMap(f)
+
+  override def tailRecM[A, B](a: A)(f: A => ZManaged[R, E, Either[A, B]]): ZManaged[R, E, B] =
+    ZManaged.succeedLazy(f(a)).flatMap(identity).flatMap {
+      case Left(nextA) => tailRecM(nextA)(f)
+      case Right(b)    => ZManaged.succeed(b)
+    }
+}
+
+private class CatsZManagedMonadError[R, E] extends CatsZManagedMonad[R, E] with MonadError[ZManaged[R, E, ?], E] {
+  override def raiseError[A](e: E): ZManaged[R, E, A] = ZManaged.fromEffect(ZIO.fail(e))
+
+  override def handleErrorWith[A](fa: ZManaged[R, E, A])(f: E => ZManaged[R, E, A]): ZManaged[R, E, A] =
```

Comment:
Would be good to have this (as `catchAll`) and `combineK` (as `orElse / <>`) on `ZManaged` itself, instead of having the logic here.

They can also be further trimmed down by implementing `ZManaged#foldM`, but that can be done separately :-)

Reply:
I'll see what I can do